From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-hugetlb-convert-free_huge_page-to-folios-fix
Date: Mon Oct 17 01:32:30 PM PDT 2022

fix build

In file included from ./include/linux/mm.h:6,
                 from mm/hugetlb.c:8:
mm/hugetlb.c: In function 'free_huge_page':
mm/hugetlb.c:1707:47: error: passing argument 1 of 'dump_page' from incompatible pointer type [-Werror=incompatible-pointer-types]
 1707 |         VM_BUG_ON_PAGE(folio_mapcount(folio), folio);
      |                                               ^~~~~
      |                                               |
      |                                               struct folio *
./include/linux/mmdebug.h:21:35: note: in definition of macro 'VM_BUG_ON_PAGE'
   21 |                         dump_page(page, "VM_BUG_ON_PAGE(" __stringify(cond)")");\
      |                                   ^~~~
./include/linux/mmdebug.h:12:29: note: expected 'struct page *' but argument is of type 'struct folio *'
   12 | void dump_page(struct page *page, const char *reason);
      |                ~~~~~~~~~~~~~^~~~
cc1: all warnings being treated as errors
make[2]: *** [scripts/Makefile.build:250: mm/hugetlb.o] Error 1
make[1]: *** [scripts/Makefile.build:500: mm] Error 2
make: *** [Makefile:1992: .] Error 2

Cc: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/hugetlb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/hugetlb.c~mm-hugetlb-convert-free_huge_page-to-folios-fix
+++ a/mm/hugetlb.c
@@ -1712,7 +1712,7 @@ void free_huge_page(struct page *page)
 	unsigned long flags;
 
 	VM_BUG_ON_FOLIO(folio_ref_count(folio), folio);
-	VM_BUG_ON_PAGE(folio_mapcount(folio), folio);
+	VM_BUG_ON_FOLIO(folio_mapcount(folio), folio);
 
 	hugetlb_set_folio_subpool(folio, NULL);
 	if (folio_test_anon(folio))
_
